.github/workflows/test.yml

```diff
@@ -53,9 +54,9 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install system dependencies
-        run: |
-          sudo apt update
-          sudo apt install --no-install-recommends -y gdal-bin
+        uses: daaku/gh-action-apt-install@49782cd9703eaa5c436b9e6caa6529e7c5e50ecd  # v4
```

Comment:
I'm not a big fan of linking to individual user's GitHub Actions with the whole supply-chain fun going around the open source world (not that I have any reason to suspect this guy has been compromised). It looks like this could be replaced with a single call to the official `actions/cache` action:

```yaml
    - id: cache
      uses: actions/cache@v4
      with:
        path: ~/.apt-cache
        key: apt-cache-${{ runner.os }}
```

Reply:
Agreed.  This is an exposure to an uncertain software layer.